src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPropertySymbol.cs

```diff
@@ -21,14 +17,29 @@ internal sealed class SynthesizedRecordPropertySymbol : SourcePropertySymbolBase
 
         public SynthesizedRecordPropertySymbol(
             SourceMemberContainerTypeSymbol containingType,
+            CSharpSyntaxNode syntax,
             ParameterSymbol backingParameter,
+            bool isOverride,
             DiagnosticBag diagnostics)
-            : base(containingType,
+            : base(
+                containingType,
                 binder: null,
-                backingParameter.GetNonNullSyntaxNode(),
+                syntax: syntax,
+                getSyntax: syntax,
+                setSyntax: syntax,
+                arrowExpression: null,
+                interfaceSpecifier: null,
+                modifiers: DeclarationModifiers.Public | (isOverride ? DeclarationModifiers.Override : DeclarationModifiers.None),
```

Comment:
>DeclarationModifiers.Public | (isOverride ? DeclarationModifiers.Override : DeclarationModifiers.None) [](start = 27, length = 102)

This lacks required validation. Please restore original implementation when validation is performed in the base constructor.